### ISSUE TITLE
[Screen Readers-Ask HR Virtual Assistant]: Screen readers(NVDA) is narrating incorrect  information in chat window when user invoked the Suggested Questionnaire button present in the “Ask HR Virtual Assistant” chat window.

### ISSUE THREAD

**Issue by KattaBhargav** [microsoft/BotFramework-WebChat#4306]
### PWD Impact:
Screen reader users are getting impacted/confused if screen readers(NVDA/JAWS) narrating hidden attachment information (which is not present in the current UI)in chat window when user invoked the Suggested Questionnaire button present in the “Ask HR Virtual Assistant” chat window. 

### Repro Steps:
Step 1: Enable Screen readers (NVDA/JAWS) and Open the above URL in the Chrome/Edge Browser using valid credentials.
Step 2: 'AskHR Virtual Assistant' Home page gets opened.
Step 3: Navigation: Select Manage positions--> select View Positions(PCNs) --> Yes--> Great! Thank you for visiting the AskHR Virtual Assistant. Please rate your experience gets displayed in chat window using down arrow key navigate to Please rate your experience content and verify the Screen readers(NVDA/JAWS) behavior 

### Actual Result:
Screen readers(NVDA) is narrating incorrect information in chat window when user invoked the Suggested Questionnaire button present in the “Ask HR Virtual Assistant” chat window.
**Ex:** NVDA narrating as "Out of activity bot attached".

### Expected Result:
Screen readers(NVDA) shouldn't narrate hidden information in chat window when user invoked the Suggested Questionnaire button present in the “Ask HR Virtual Assistant” chat window.
**Ex:** NVDA should  narrate as "Great! Thank you for visiting the AskHR Virtual Assistant Please rate your experience Great! Please rate your experience button Terrible Rate 1" 

### MAS Reference:
[MAS 1.3.2 – Meaningful Sequence](https://microsoft.sharepoint.com/:w:/r/sites/accessibility/_layouts/15/Doc.aspx?sourcedoc=%7b889A51B3-CAA2-4419-87D4-09BFAD2E8379%7d&file=MAS%201.3.2%20%E2%80%93%20Meaningful%20Sequence%20.docx&wdLOR=cE46C6224-661F-4808-8CF1-053E77173597&action=default&mobileredirect=true) 

### WCAG Reference:
https://www.w3.org/WAI/WCAG21/Understanding/meaningful-sequence.html

### Test Environment:

- OS: Windows 11 version 22H2 (OS Build 22621.1)
- Browser: Microsoft Edge Version 104.0.1271.2 (Official build) dev (64-bit)
- URL: [AskHR Virtual Assistant (azureedge.net)](https://askhrvadev.azureedge.net/?botOrigin=ec)
- Tools: NVDA

**Comment by KattaBhargav** [comment]
#A11yMAS;#A11ySev2;#HCL;#WCAG1.3.2;#Accessibility;#Win11-Edge(Chromium);#BotFrameworkWebChat-Aug21;#Linked:Bug8012704;#LinkedBug;

**Comment by KattaBhargav** [comment]
Bug copied from Azurebot team Bug id:-[BUG 8012704](https://microsoftit.visualstudio.com/OneITVSO/_workitems/edit/8012704)

**Comment by compulim** [comment]
We designed the narration with the following patterns. You can read ([full details in this article](https://github.com/microsoft/BotFramework-WebChat/blob/main/docs/ACCESSIBILITY.md#ux-narration-of-messages)).

- When a message with Adaptive Card arrive, screen reader should narrate the message along with card content
- When navigating using <kbd>UP</kbd>/<kbd>DOWN</kbd> arrow keys, screen reader should NOT narrate the card content until the customer click to "enter" into the activity

We talked with Prateek offline. His suggestion is to change the role description from "article" (built-in) to "message". So JAWS will change its narration and it will be more intuitive for the customer to understand the message.

- From "start article, bot said, hello, end article"
- To "start message, bot said, hello, end message"

However, Apple VoiceOver and Android TalkBack does not work with role description consistently. Especially, it negatively impact customers using Safari. So we prefer not to fix this issue until VoiceOver/TalkBack has fixed their issues. My investigations can be found in the links below:

- https://github.com/compulim/experiment-accessibility-quirks#ios-safari--voiceover-aria-roledescription-will-mute-the-role-description
- https://github.com/compulim/experiment-accessibility-quirks#ios-safari--voiceover-aria-label-seems-mistreated-as-aria-roledescription

So until VoiceOver/TalkBack has their issues fixed, we cannot proceed further without affecting VoiceOver-specific customers.

**Comment by mishraadi** [comment]
Verified the issue on OS: Windows 11 version 22H2 (OS Build 22621.1000)
Browser: Microsoft Edge Version 104.0.1271.2 (Official build) dev (64-bit)
URL: [AskHR Virtual Assistant (azureedge.net)](https://askhrvadev.azureedge.net/?botOrigin=ec)
Tools: NVDA
and the issue doesn't repro anymore. Proper expected information is announced. So, closing it.
